pkg/smokescreen/smokescreen.go

```diff
@@ -1216,11 +1229,23 @@ func checkACLsForRequest(config *Config, req *http.Request, destination hostport
 		return decision
 	}
 
-	role, roleErr := getRole(config, req)
-	if roleErr != nil {
-		config.MetricsClient.Incr("acl.role_not_determined", 1)
-		decision.Reason = "Client role cannot be determined"
-		return decision
+	var role string
+	var roleErr error
+	
+	// Check if role is already populated in SmokescreenContext (e.g., from CONNECT in MITM mode)
+	if sctx.Decision != nil && sctx.Decision.Role != "" && sctx.isConnectMitm {
+		role = sctx.Decision.Role
```

Comment:
Nit: should we be a bit more defensive here? Something like:
```suggestion
	if sctx.isConnectMitm {
		if sctx.Decision == nil || sctx.Decision.Role == "" {
			config.Log.WithFields(logrus.Fields{
				"decision_nil":  sctx.Decision == nil,
				"role_empty":    sctx.Decision != nil && sctx.Decision.Role == "",
			}).Error("MITM request missing required role from CONNECT phase")
			config.MetricsClient.Incr("acl.role_not_determined", 1)
			decision.Reason = "Client role cannot be determined"
			return decision
		}
		role = sctx.Decision.Role
```